scripts/recursive_build.py

```diff
@@ -1,119 +1,85 @@
-import json
-import requests
-import py7zr
-import io
-import os
-import re
-import sqlite3
-import subprocess
-
-# Path to the necto_db.db file.
-dbPath = '/home/runner/.MIKROE/NECTOStudio7/databases/necto_db.db'
+import os, re, subprocess, shutil, json, sqlite3
+
+from packaging import version
+
+# Global variable for local_app_data_path
+local_app_data_path = '/home/runner/.MIKROE/NECTOStudio7'
 
 # Path for storing artifacts.
 testPath = '/home/runner/test_results'
 
 # Path to sdk_build_automation tool.
-toolPath = '/home/runner/MikroElektronika/NECTOStudio/bin'
+toolPath = '/home/runner/MikroElektronika/NECTOStudio/bin/sdk_build_automation'
 
 # Global variable to trace failed tests.
 build_failed = False
 
 # Supported compilers list for each architecture.
 compiler_list = {
-    'ARM': ['gcc_arm_none_eabi', 'clang-llvm', 'mikrocarm'],
+    'ARM': ['gcc_arm_none_eabi', 'clang-llvm'],
     'RISCV': ['xpack-riscv-none-embed-gcc', 'clang-llvm-riscv'],
-    'PIC': ['mchp_xc8', 'mikrocpic'],
-    'DSPIC': ['mchp_xc16', 'mikrocdspic'],
-    'PIC32': ['mchp_xc32', 'mikrocpic32'],
-    'AVR': ['mikrocavr']
+    'PIC': ['mchp_xc8'],
+    'DSPIC': ['mchp_xc16'],
+    'PIC32': ['mchp_xc32']
 }
 
-# Repo information.
-REPO_OWNER = 'MikroElektronika'
-REPO_NAME = 'core_packages'
-
-# Returns the list of compilers based on the given name and type.
-def get_compilers(name):
-    if any(substring in name for substring in ["ATSAM", "STM", "TM4C", "MK"]):
-        return compiler_list["ARM"], "ARM"
-    elif "GD32" in name:
-        return compiler_list["RISCV"], "RISCV"
-    elif "PIC32" in name:
-        return compiler_list["PIC32"], "PIC32"
-    elif any(substring in name for substring in ["DSPIC", "PIC24", "dsPIC"]):
-        return compiler_list["DSPIC"], "DSPIC"
-    elif any(substring in name for substring in ["PIC18", "PIC16", "PIC12", "PIC10"]):
-        return compiler_list["PIC"], "PIC"
-    elif "AT" in name and "ATSAM" not in name:
-        return compiler_list["AVR"], "AVR"
-    
-def get_all_mcus(sdk_version):
-    mcu_list = []
-    conn = sqlite3.connect(dbPath)
-    cursor = conn.cursor()
-
-    cursor.execute(f"""
-        SELECT SDKToDevice.device_uid
-        FROM SDKToDevice
-        INNER JOIN Devices ON SDKToDevice.device_uid = Devices.uid
-        WHERE SDKToDevice.sdk_uid = '{sdk_version}'
-        AND SDKToDevice.device_uid NOT LIKE '%\\_%' ESCAPE '\\';
-    """)
-    rows = cursor.fetchall()
-    if rows:
-        mcu_list.extend([row[0] for row in rows])
-    mcu_list[:] = sorted(mcu_list)
-
-    conn.close()
-
-    return mcu_list
+# Define a REGEXP function for SQLite.
+def regexp(expr, item):
```

Comment:
Do we need this at all?

It seems to me as if `functionRegex` will work just fine.

If so, leave just one.

Reply:
Replaced regexp with functionRegex